src/worker.cpp

```diff
@@ -1159,50 +1159,27 @@ types::VPDMapVariant Worker::parseVpdFile(const std::string& i_vpdFilePath)
             "Empty VPD file path passed to Worker::parseVpdFile. Abort processing");
     }
 
-    bool l_isPostFailActionRequired = false;
-
-    // check if the FRU qualifies for pre action.
-    if (jsonUtility::isActionRequired(m_parsedJson, i_vpdFilePath, "preAction",
-                                      "collection"))
+    try
```

Comment:
kept try&catch block outside,
so if either PreAction fails or Vpd Prasing fails, just execute PostFailAction of present in the JSON for that FRU.

Reply:
I got that, I was trying to say you could have just modified one line this condition

`if (processPreAction(i_vpdFilePath, "collection") &&
            jsonUtility::isActionRequired(m_parsedJson, i_vpdFilePath,
                                          "PostFailAction", "collection"))
        {
            l_isPostFailActionRequired = true;
        }`

to 

`if (processPreAction(i_vpdFilePath, "collection"))
        {
           if(jsonUtility::isActionRequired(m_parsedJson, i_vpdFilePath,
                                          "PostFailAction", "collection"))
           {
                 l_isPostFailActionRequired = true;
           }
        }`

and that would have solved the concern here, you need not do other changes like removing boolean variable and move code.

Just trying to avoid too many code changes at this point.
Anyways, if you have tested sufficiently, even this change works for me.

Reply:
if pre Action fails why will post fail action not be triggered? It should be triggered , I had asked for these changes.

Reply:
> if pre Action fails why will post fail action not be triggered? It should be triggered , I had asked for these changes.

Not all FRUs having "preAction" has "post fail action" defined as of now, so not mandatory that if pre Action fails post fail action should get trigerred.

Anyways, this change is anyways not solving that case, it is solving situation where if "post fail action" is not defined then the code was marking preAction passed also as preAction failed.

Reply:
nope I think there is some misunderstanding here, we need all cases done , post fail action of available needs to run on precaution fail due to the fact that reaction may have multiple actions internally and some may have passed and those need to be set to offline mode.

The change is solving that case also , could you tell me exactly which case is nothing solved here

Reply:
> post fail action of available needs to run on precaution fail due to the fact that reaction may have multiple actions internally and some may have passed and those need to be set to offline mode.

That was being done before as well. There was no issue in that flow. post fail action was getting executed for all the FRUs for which it was available in case of failure.

This change only fixes the flow for FRUs which does not have post fail action defined, and this fix is necessary else collection of those FRUs will fail.

Reply:
> That was being done before as well. There was no issue in that flow. 

Got It, that && condition was failing and hence the boolean was not being set, hence the post fail action was not getting executed for exception case as well.

Reply:
I guess the confusion is because:

In worker.cpp flow, execution of post fail action is done only in case pre-action passes and then later there is some problem in collection of the FRU, where as error for error scenarios "post fail action" is getting executed at that utility function itself.

eg: if there is any problem in setting GPIO. post fail action will execute in the body of that function itself.
https://github.com/ibm-openbmc/openpower-vpd-parser/blob/P11_Dev/include/utility/json_utility.hpp#L459

Now, in case you want both good and bad case handling in worker.cpp flow itself then @branupama please remove the execution of post fail action from the respective API whose link I have posted above.

Reply:
Done. 
And also removed check for tag "PostFailAction" in executePostFailAction() API, as we are checking the tag outside itself before calling this API.

Reply:
Why would you removed the tag? I asked you to remove the execution of complete failure part from there as that is being handled in pre-action flow.